rules-reviewed/jboss-eap/uncategorized/environment-dependent.windup.xml

```diff
@@ -64,19 +63,37 @@
         </rule>
         <rule id="environment-dependent-calls-03000">
             <when>
-                <javaclass references="javax.naming.InitialContext({*})" as="default">
+                <javaclass references="javax.naming.InitialContext(java.util.Hashtable{*}" as="default">
                     <location>CONSTRUCTOR_CALL</location>
                 </javaclass>
             </when>
             <perform>
                 <iteration>
-                    <hint title="InitialContext initialization" effort="0" severity="optional">
-                    <message>InitialContext looks for the properties located in multiple places. Besides checking the map that may be passed to this constructor, it also checks the system properties and jndi.properties file. Ensure that these InitialContext connection properties do not need to be changed for JBoss.</message>
+                    <hint title="Proprietary InitialContext initialization" effort="1" severity="mandatory">
+                    <message>In JBoss EAP, the InitialContext should be instantiated with no arguments. Once an instance is constructed, look up the service using portable JNDI lookup syntax. Ensure also that in case system properties for InitialContext are provided, they do not need to be changed for the JBoss.
+
+                        `InitialContext context = new InitialContext();
+                        Service service = (Service) context.lookup( "java:app/service/" + ServiceImpl.class.getSimpleName() );`
+                        </message>
+                  <!-- <link href="https://access.redhat.com/articles/1496973" title="Migrate Oracle WebLogic Applications to Use Portable JNDI Syntax in Red Hat JBoss Enterprise Application Platform" /> -->
```

Comment:
here is the link to weblogic specific migration article that is however quite generic